AEXS/aex-141.md

```diff
@@ -300,124 +314,111 @@ ApprovalForAll(address, address, string)
 
-```sophia
-stateful entrypoint swap() : unit
-```
+## Extension Mintable ("mintable")
 
-| parameter | type |
-| :--- | :--- |
+The `mintable` extension SHOULD be used for generic NFT minting without specific requirements in regards to minting.
 
-| return | type |
-| :--- | :--- |
-| () | unit |
+### mint\(\)
+
+Issues a new token to the provided address. If the `owner` is a contract, `IAEX141Receiver.on_aex141_received` will be called with `data` if provided.
 
-### check_swap\(\)
+Emits the `Mint` event.
 
-Returns the amount of NFTs that were burned through `swap` for the provided account. 
+Throws if the call to `IAEX141Receiver.on_aex141_received` implementation failed (safe transfer)
 
 ```sophia
-stateful entrypoint check_swap(account: address) : int
+stateful entrypoint mint(owner: address, metadata: option(metadata), data: option(string)) : int
 ```
 
 | parameter | type |
 | :--- | :--- |
-| account | address |
+| owner | address |
+| metadata | option(metadata) |
+| data  | option(string) |
 
 | return | type |
 | :--- | :--- |
-| int | int |
-
-### swapped\(\)
+| token_id | int |
 
-Returns all of the swapped NFTs that are stored in contract state. 
+## Extension Burnable ("burnable")
 
-```sophia
-stateful entrypoint swapped() : map(address, int)
-```
+The `burnable` extension SHOULD be used if NFTs within a contract are intended to be burnable.
 
-| return | type |
-| :--- | :--- |
-| swapped | map(address, int) |
+### burn\(\)
 
-## Events
+Burns the NFT with the provided `token_id`.
 
-**Swap** - MUST trigger when tokens are swapped using the `swap` function.
+Emits the `Burn` event.
 
-The swap event arguments should be as follows: `(account,  value)`
+Throws if `Call.caller` is NOT the current owner or NOT approved to transfer on behalf of the owner.
 
 ```sophia
-Swap(address, int)
+stateful entrypoint burn(token_id: int) : unit
 ```
 
 | parameter | type |
 | :--- | :--- |
-| account| address |
-| value | int |
+| token_id | int |
 
-# Receiver contract interface
+## Extension Events
 
-## NFTReceiver
+**Mint**
+
+This event MUST be triggered whenever a new token is minted with the `mintable` extension.
+
+The event arguments should be as follows: `(to, token_id)`
 
 ```sophia
-contract interface NFTReceiver = 
-    entrypoint on_nft_received : (address, address, int, option(string)) => bool
+Mint(address, int)
 ```
 
-### on_nft_received\(\)
+| parameter | type |
+| :--- | :--- |
+| to | address |
+| token_id | int |
 
-Deals with receiving NFTs on behalf of a contract. Contracts MUST implement this interface to be able to receive NFTs. Mint and transfer transactions will invoke the `on_nft_received` function.
+**Burn** - MUST trigger when NFTs are burned using the `burn` function.
```

Comment:
if the Burn event usage is limited to the "burn" function  it can't be used in indexing, every burn no matter where or how it is done MUST trigger the event in order to allow that

Reply:
fixed: https://github.com/aeternity/AEXs/commit/121c4505c8e051fb26314a29aa58902fa79a0826

Reply:
update again due to typo and formatting: https://github.com/aeternity/AEXs/commit/f3d54075fa363d953ea5a1c742b62599471e3b88

Reply:
looks good